File: php/phpsearch/src/phpsearch/SearchResult.php
<?php declare(strict_types=1);

namespace phpsearch;

use phpfind\FileResult;

/**
 * Class SearchResult
 */
class SearchResult
{
    public string $pattern;
    public ?FileResult $file;
    public int $line_num;
    public int $match_start_index;
    public int $match_end_index;
    public ?string $line;
    public array $lines_before;
    public array $lines_after;

    public function __construct(
        string $pattern,
        ?FileResult $file,
        int $line_num,
        int $match_start_index,
        int $match_end_index,
        ?string $line,
        array $lines_before,
        array $lines_after
    ) {
        $this->pattern = $pattern;
        $this->file = $file;
        $this->line_num = $line_num;
        $this->match_start_index = $match_start_index;
        $this->match_end_index = $match_end_index;
        $this->line = $line;
        $this->lines_before = $lines_before;
        $this->lines_after = $lines_after;
    }
}

File: fsharp/FsSearch/FsSearchLib/SearchResult.fs
namespace FsSearchLib

open System.Text.RegularExpressions
open FsFindLib

module SearchResult =
    type t = {
        SearchPattern : Regex;
        File : FileResult.t;
        LineNum : int;
        MatchStartIndex : int;
        MatchEndIndex : int;
        Line : string;
        LinesBefore : string list;
        LinesAfter : string list;
    }

    let Create (pattern : Regex) (lineNum : int) (startIndex : int) (endIndex : int) (line : string) (linesBefore : string list) (linesAfter : string list) : t =
        {
            SearchPattern = pattern;
            File = { Containers=[]; File=null; FileType=FileType.Unknown; };
            LineNum = lineNum;
            MatchStartIndex = startIndex;
            MatchEndIndex = endIndex;
            Line = line;
            LinesBefore = linesBefore;
            LinesAfter = linesAfter;
        }
;;
